src/pep8ext_naming.py

```diff
@@ -466,13 +465,15 @@ def _extract_names(assignment_target):
             elif not PY2 and element_type is ast.Starred:  # PEP 3132
                 for n in _extract_names(element.value):
                     yield n
-        return
-    if target_type is ast.ExceptHandler:
+    elif target_type is ast.ExceptHandler:
         if PY2:
-            yield assignment_target.name.id
+            if isinstance(assignment_target.name, ast.Tuple):
```

Comment:
We could also just call `_extract_names` here recursively:

```python
if PY2:
    for n in _extract_names(assignment_target.name):
        yield n
else:
    yield assignment_target.name.id
```

That code is more succinct but will run a little slower (because more stack frames). That's probably fine in this case because I would assume `except` statements are relatively rare out of all visited nodes.

Reply:
Although this isn't worse in any sense so we may as well just stick with this I guess?

Reply:
I thought about it a little more. If we go with the recursive approach, it's going to deserve a comment explaining why it's so different than the Python 3 path. If we leave the code explicit (as above), the difference and intent are much more obvious.

Reply:
Agreed

Reply:
Although a comment _still_ would be a good idea anyway